### ISSUE TITLE
Show duplicative heading level text when showIconLabels is enabled

### ISSUE THREAD

**Issue by richtabor** [WordPress/gutenberg#56621]
As a follow-up to https://github.com/WordPress/gutenberg/pull/56035, instead of adding the duplicative visible text of `Heading $level` next to the corresponding `H $level` icon, let's enable the visible text when `showIconLabels` is enabled. 

![CleanShot 2023-11-28 at 19 13 33](https://github.com/WordPress/gutenberg/assets/1813435/6af6984c-0d73-437a-9717-8dba81c55963)

**Comment by carolinan** [comment]
I disagree and I don't consider this a "follow-up", but a suggestion to revert the result of the discussion.
The benefit of visible text has been clarified in https://github.com/WordPress/gutenberg/pull/56035
